containerd-shim-v2/service.go

```diff
@@ -65,6 +65,12 @@ func New(ctx context.Context, id string, publisher events.Publisher) (cdshim.Shi
 	// it will output into stdio, from which containerd would like
 	// to get the shim's socket address.
 	logrus.SetOutput(ioutil.Discard)
+	opts := ctx.Value(cdshim.OptsKey{}).(cdshim.Opts)
```

Comment:
Is this setting needed? Since the log has been discarded by "logrus.SetOutput(ioutil.Discard)" previously.

Reply:
Yes,  the followed ` katautils.SetLogger` will use the level.

Reply:
I mean since the log's output has been discarded, thus whatever the log level you set, you wouldn't get what you want. Thus, is better to remove the setting here?

Reply:
They are not the same `logger`, maybe I change the SetLevel line to 
```
logger.Logger.SetLevel(logrus.DebugLevel)
```
will make it clearly?

Reply:
In my opinion, they are the same logger which is the standard logger, and it will be reset the level here https://github.com/containerd/containerd/blob/master/runtime/v2/shim/shim.go#L133, thus I think there is no need 
to set the level here.

Reply:
[startShim](https://github.com/containerd/containerd/blob/master/runtime/v2/shim/shim.go#L211) will start a new process  [here](https://github.com/kata-containers/runtime/blob/master/containerd-shim-v2/service.go#L201), do you think the level will effect new process? You can build and try my patch, and you will trust me .

Reply:
If `debugFlag` isn't set, the level will not be reset. You can build and test my patch, and you will trust me.

Reply:
If the debugFlag isn't set, then it would use the default level InfoLevel. So here you want to change it to WarnLevel
if the debugFlag isn't set?

Reply:
Yes, too many INFO logs are generated by shim. If anyone need more detail logs, he can just turn debug on.

Reply:
In that case, is it better to change it as below:

if !opts.Debug {
       logrus.SetLevel(logrus.WarnLevel)
}